sos/plugins/salt.py

```diff
@@ -0,0 +1,41 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+
+
+class Salt(Plugin, RedHatPlugin, UbuntuPlugin):
+    """Salt
+    """
+
+    plugin_name = 'salt'
+    profiles = ('storage')
+
+    option_list = [
+        ("log", "gathers all salt logs", "slow", False)
+    ]
+
+    packages = (
+        'salt',
+    )
+
+    def setup(self):
+        self.add_copy_spec([
+            "/etc/salt/",
+            "/var/log/salt/"
```

Comment:
Looks like this is missing something to implement the `logs` option defined above? Normally we collect a size-limited snapshot of the first (current/non-rotated) log file by default, with size limits and additional log collection determined by plugin or global options.

See other plugins, or the wiki entries, for examples.